File: SWTOR HoloNetTests/Extension/URLComponentsTests.swift
//
//  URLComponentsTests.swift
//  SWTOR HoloNet
//
//  Created by Ivan Fabijanovic on 04/11/14.
//  Copyright (c) 2014 Ivan Fabijanović. All rights reserved.
//

import UIKit
import XCTest

class URLComponentsTests: XCTestCase {
    func testQueryValueForName_Success() {
        let url = "http://www.holonet.test?param=value"
        let value = URLComponents(string: url)!.queryValue(name: "param")!
        
        XCTAssertEqual(value, "value", "")
    }
    
    func testQueryValueForName_InvalidUrl() {
        let url = "invalid url string"
        let components = URLComponents(string: url)
        
        XCTAssertNil(components, "")
    }
    
    func testQueryValueForName_NoParameters() {
        let url = "http://www.holonet.test"
        let value = URLComponents(string: url)!.queryValue(name: "param")
        
        XCTAssertNil(value, "")
    }
    
    func testQueryValueForName_MissingParameter() {
        let url = "http://www.holonet.test?param=value"
        let value = URLComponents(string: url)!.queryValue(name: "otherParam")
        
        XCTAssertNil(value, "")
    }
    
    func testQueryValueForName_MultipleParameters() {
        let url = "http://www.holonet.test?param1=value1&param2=value2"
        let value1 = URLComponents(string: url)!.queryValue(name: "param1")!
        let value2 = URLComponents(string: url)!.queryValue(name: "param2")!
        
        XCTAssertEqual(value1, "value1", "")
        XCTAssertEqual(value2, "value2", "")
    }
}

File: android/app/src/test/java/com/ifabijanovic/holonet/helper/URLComponentsTests.kt
package com.ifabijanovic.holonet.helper

import org.junit.Assert.*
import org.junit.Test

/**
 * Created by feb on 19/03/2017.
 */
class URLComponentsTests {
    @Test
    fun queryValue_Success() {
        val url = "http://www.holonet.test?param=value"
        val value = URLComponents(url).queryValue("param")

        assertNotNull(value)
        assertEquals("value", value)
    }

    @Test
    fun queryValue_NoParameters() {
        val url = "http://www.holonet.test"
        val value = URLComponents(url).queryValue("param")

        assertNull(value)
    }

    @Test
    fun queryValue_MissingParameter() {
        val url = "http://www.holonet.test?param=value"
        val value = URLComponents(url).queryValue("otherParam")

        assertNull(value)
    }

    @Test
    fun queryValue_MultipleParameters() {
        val url = "http://www.holonet.test?param1=value1&param2=value2"
        val value1 = URLComponents(url).queryValue("param1")
        val value2 = URLComponents(url).queryValue("param2")

        assertNotNull(value1)
        assertEquals("value1", value1)
        assertNotNull(value2)
        assertEquals("value2", value2)
    }

    @Test
    fun queryValue_MalformedUrl() {
        val url = "somewhere/test?param=value"
        val value = URLComponents(url).queryValue("param")

        assertNotNull(value)
        assertEquals("value", value)
    }
}
